### PR TITLE
Add test coverage for the gaps surfaced by cpp-audit (#493)

### LINKED ISSUE #493: Test coverage gaps surfaced by cpp-audit (apply_to_grove; CLI negative/edge e2e)
From `/cpp-audit`. Missing negative/edge coverage for otherwise-tested surfaces.

- [ ] `tests/cli/links-test.cpp` — `links::apply_to_grove` (name resolution, edge dedup, metadata-as-key, missing-name throw, and its documented `edges_added` return) is covered only indirectly via subprocess e2e; the return count is never asserted anywhere. Add a direct unit test building a small grove + name_map, asserting the edge count and the missing-name `std::runtime_error`.
- [ ] `tests/cli/` — no negative test for `intersect`'s unsupported-query-format rejection (`intersect.cpp:167-171`), the only CLI error branch without one. Feed a FASTA/plain-text query to `isec` and assert non-zero exit + message.
- [ ] `tests/cli/` — no e2e test for `idx` on an empty (0-record) input, though the readers support the empty-file contract and a `tests/io/data/empty.bed` fixture now exists. Assert a valid deserializable 0-record `.gg`.
- [ ] `tests/cli/` — no CLI-level test feeding a corrupt / wrong-magic file to `isec -i` (unit-covered in `gg_format_test.cpp`, but the CLI's clean-error + exit-code translation is untested).

### PR DESCRIPTION
**Pull request by riasc**
Closes #493. Test-only — no source changes.

Fills the four negative/edge coverage gaps the audit surfaced.

## What's added
- **`links::apply_to_grove` direct unit tests** (`tests/cli/links-test.cpp`) — previously exercised only indirectly through subprocess e2e, and the `edges_added` return was never asserted. New tests build a small grove + name_map and check:
  - `resolvesNamesDedupsAndCountsEdges` — an exact-duplicate row is collapsed while the same pair with two different metadata values is kept as two parallel edges (`added == 3`).
  - `throwsOnNameNotInMap` — an unresolved name throws `std::runtime_error` naming the missing name.
- **`intersect` negative e2e** (`tests/cli/intersect-e2e-test.cpp`):
  - `UnsupportedQueryFormatRejected` — a FASTA query is rejected (`must be BED…`), exit ≠ 0.
  - `CorruptIndexFileRejected` — a non-`.gg` file passed to `-i` fails the header magic check cleanly (`magic mismatch`), no crash.
- **`idx` empty-input e2e** (`tests/cli/index-e2e-test.cpp`):
  - `IndexEmptyInputProducesDeserializableIndex` — indexing an empty (0-record) BED succeeds and writes a deserializable `.gg`, proven by then querying it in place via `isec -i` (exit 0, no overlaps).

## Verification
- `apply_to_grove` logic verified out of band with a standalone harness: `added == 3`, and the missing-name throw names the name.
- All three test files syntax-check clean against the real gtest/htslib headers.
- The CLI e2e assertions are matched to the exact code messages (`intersect.cpp:170`, `gg_format.cpp:74`); CI runs them against a freshly built binary (a stale local binary was wrong-arch, so CI is the e2e check).

## Test plan
- [ ] I, as a human being, have checked each line of code
- [x] `LinksApply.resolvesNamesDedupsAndCountsEdges` / `throwsOnNameNotInMap`
- [x] `CLIIntersectE2ETest.UnsupportedQueryFormatRejected` / `CorruptIndexFileRejected`
- [x] `CLIIndexE2ETest.IndexEmptyInputProducesDeserializableIndex`

## Summary by CodeRabbit

* **Tests**
  * Added coverage confirming empty BED inputs can be indexed and queried successfully.
  * Added validation for rejecting unsupported FASTA query files with clear error messages.
  * Added checks that invalid or corrupted index files are rejected safely.
  * Added coverage ensuring duplicate links are handled correctly while distinct metadata is preserved.
  * Added validation for clear errors when links reference unknown entries.

### CODE REVIEW

File: tests/cli/index-e2e-test.cpp [hunk 1]
```diff
@@ -575,4 +575,25 @@ TEST_F(CLIIndexE2ETest, InvalidOptionValueFailsCleanly) {
...
     auto result = run_command(cli("idx \"" + target_path.string() + "\" -k notanumber"));
     EXPECT_EQ(result.exit_code, 1) << result.output;
+}
+
+TEST_F(CLIIndexE2ETest, IndexEmptyInputProducesDeserializableIndex) {
+    // An empty (0-record) BED is a valid input: idx must succeed and write a
+    // deserializable .gg. Proven by then querying it in place (isec -i), which
+    // deserializes the index and returns no overlaps.
+    fs::path empty_bed = fs::temp_directory_path() / "genogrove_idx_empty.bed";
+    { std::ofstream out(empty_bed); }  // zero bytes
```
[the remark this answers is not in the document]
Comment (riasc) [reply tests/cli/index-e2e-test.cpp:585]: **Verdict: Skip** (see the summary comment for the full rationale).

The entire existing e2e suite uses fixed temp filenames in `temp_directory_path()` with manual `fs::remove` cleanup — e.g. `genogrove_idx_test.gg`, `genogrove_e2e_test_output.bed`, `temp_invalid_format.bed`, `temp_invalid_coords.bed`, and ~30 others. These three additions deliberately follow that established convention.

The symlink-race (CWE-377) is a real category, but for a library's CI test fixtures the threat is negligible (isolated CI runners; a dev-machine attacker would need local access and a race), and hardening only 3 of the suite's many temp-file sites would be inconsistent and give false assurance. If temp-file hardening is wanted it should be one shared helper applied suite-wide — a separate change, not this test-coverage PR. Happy to file that as its own issue if you'd like.

**Comment by riasc** [comment]
## Code Review

Test-only PR; reviewed for correctness of the assertions and coverage value. CI is green on all 7 configs.

**Correctness — clean.** The `apply_to_grove` unit tests exercise the real edge-dedup logic (exact-duplicate collapse vs. parallel-metadata edges → `added == 3`) and the missing-name throw; I verified this out of band with a standalone harness. The CLI e2e assertions are matched to the exact source messages (`must be BED` → `intersect.cpp:170`, `magic mismatch` → `gg_format.cpp:74`), and the empty-`idx` test proves deserializability via an `isec -i` round-trip. No issues found.

### CodeRabbit comments — assessed
- **Predictable temp filenames / symlink race (CWE-377), inline on 3 sites** — **Skip.** The whole e2e suite uses fixed temp names + manual `fs::remove`; these additions follow that convention. The race is negligible for CI fixtures, and hardening 3 of ~30 sites is inconsistent. A shared temp-file helper applied suite-wide would be a separate change — I can file it as its own issue if wanted.
- **`ASSERT_EQ` + RAII guard on the empty-`idx` test** — **Partial.** Switching the `idx` step to `ASSERT_EQ` (fail-fast) is worth doing; the RAII guard I'll skip (the fixture's `TearDown` already removes `tmp_output`, and a one-off guard is inconsistent with the suite).
- **RAII `temp_tsv` helper in links-test** — **Skip.** Same convention rationale; the manual-cleanup pattern is used throughout the suite.

No bugs, no guideline violations.

---

## Update — round 1

| Item | Status | Notes |
|------|--------|-------|
| `EXPECT_EQ`→`ASSERT_EQ` on idx step (empty-index test) | ✅ Addressed | Fail fast; don't isec a stale index (see latest commit). |
| Predictable temp names / symlink race (3 sites) | ⏭️ Skipped | Matches the suite-wide fixed-name convention; negligible for CI fixtures. Suite-wide temp helper would be a separate change. |
| RAII guards (index-e2e + links-test) | ⏭️ Skipped | `TearDown` covers `tmp_output`; one-off guards are inconsistent with the suite. |
